overlord/overlord_test.go

```diff
@@ -618,6 +618,95 @@ func (ovs *overlordSuite) TestEnsureLoopPruneRunsMultipleTimes(c *C) {
 	c.Assert(err, IsNil)
 }
 
+func (ovs *overlordSuite) testEnsureLoopPruneAfterSeeding(c *C, seeded bool, spawnTime, readyTime, currentTime time.Time) *state.State {
+	restoreIntv := overlord.MockPruneInterval(100*time.Millisecond, 1000*time.Millisecond, 1*time.Hour)
+	defer restoreIntv()
+
+	o, err := overlord.New(nil)
+	c.Assert(err, IsNil)
+
+	if seeded {
+		markSeeded(o)
+	}
+
+	st := o.State()
+	st.Lock()
+
+	restoreTimeNow := state.MockTime(spawnTime)
+	defer restoreTimeNow()
+
+	// task created with spawn time of a month ago
+	t1 := st.NewTask("foo", "...")
+	chg1 := st.NewChange("seed", "...")
+	chg1.AddTask(t1)
+
+	t2 := st.NewTask("bar", "...")
+	chg2 := st.NewChange("other-change", "...")
+	chg2.AddTask(t2)
+
+	state.MockTime(readyTime)
+	if seeded {
+		t1.SetStatus(state.DoneStatus)
+	} else {
+		t1.SetStatus(state.DoingStatus)
+	}
+	t2.SetStatus(state.DoneStatus)
+
+	// sanity
+	c.Check(st.Changes(), HasLen, 2)
+	st.Unlock()
+
+	// setting time to currentTime, affects "now" time in Prune().
+	state.MockTime(currentTime)
+
+	// start the loop that runs the prune ticker
+	o.Loop()
+
+	time.Sleep(1500 * time.Millisecond)
```

Comment:
Is there a way to avoid a 1.5s sleep in tests? Could we do it so that we get a notification that the relevant change happened?

Reply:
The only way to avoid this would be to call state.Prune() directly, but in all the tests in this package we want to do this indirectly via Loop which uses the mocked prune interval to be as close to real thing as possible.